assets/src/edit-story/components/canvas/canvasUploadDropTarget.js

```diff
@@ -48,18 +49,91 @@ const PageAreaCover = styled(PageArea)`
 `;
 
 function CanvasUploadDropTarget({ children }) {
-  const { uploadFile } = useUploader();
+  const { uploadFile, sizeErrorMessage, validErrorMessage } = useUploader();
+  const { createSnackbar } = useSnackbar();
   const insertElement = useInsertElement();
   const onDropHandler = useCallback(
-    (files) => {
-      files.forEach((file) => {
-        uploadFile(file).then((res) => {
+    async (files) => {
+      const possibleRetryFiles = [];
+      const sizeErrorFiles = [];
+      const validErrorFiles = [];
+      const otherErrorFiles = [];
+      const isMultiple = files.length > 1;
+      for (const file of files) {
+        try {
+          // eslint-disable-next-line no-await-in-loop
+          const res = await uploadFile(file);
           const resource = getResourceFromUploadAPI(res);
           insertElement(resource.type, { resource });
+        } catch (e) {
+          if (e.name !== 'SizeError' && e.name !== 'ValidError') {
+            possibleRetryFiles.push(file);
+            e.message = __('Sorry, file has failed to upload', 'web-stories');
+            e.file = file.name;
+          }
+          if (!isMultiple) {
```

Comment:
This `if` is in a wrong place. We should just have a single real processor for all the errors.

Reply:
Will be addressed in another PR